From c91d794a70d36a5c016a889708b4623e98322bc0 Mon Sep 17 00:00:00 2001
From: Nicolas Dufresne <nicolas.dufresne@collabora.com>
Date: Wed, 14 Jul 2021 16:09:41 -0400
Subject: [PATCH 1/2] gl: x11: Issue XSync to close our top level window

This is similar action as when the window handle is modified, we now issue
XSync whenever we destroy our internal window. This ensure that the window is
properly closed before the connecgtion is dropped.

Fixes #815

Part-of: <https://gitlab.freedesktop.org/gstreamer/gst-plugins-base/-/merge_requests/1226>
---
 gst-libs/gst/gl/x11/gstglwindow_x11.c | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/gst-libs/gst/gl/x11/gstglwindow_x11.c b/gst-libs/gst/gl/x11/gstglwindow_x11.c
index 960d39fb..7ee1deda 100644
--- a/gst-libs/gst/gl/x11/gstglwindow_x11.c
+++ b/gst-libs/gst/gl/x11/gstglwindow_x11.c
@@ -271,6 +271,9 @@ gst_gl_window_x11_close (GstGLWindow * window)
       XUnmapWindow (window_x11->device, window_x11->internal_win_id);
 
       XDestroyWindow (window_x11->device, window_x11->internal_win_id);
+
+      /* Ensure everything is sent immediatly */
+      XSync (window_x11->device, FALSE);
     }
     XFree (window_x11->visual_info);
 
-- 
2.30.2

